### PR TITLE
added multiplier for drops

### PR DESCRIPTION
**Pull request by gpolcode**

### CODE REVIEW

File: ValheimPlus/Drops.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                if (!__instance.m_nview.IsOwner())
+                {
+                    return true;
+                }
+                if (__instance.m_picked)
+                {
+                    return true;
+                }
+                int num = 0;
+                int num2 = Configuration.Current.Drops.baseIncreasedDropMultiplier - 1;
```
Comment (radmint) [inline ValheimPlus/Drops.cs:65]: I think it'd probably be good to restrict this multiplier to only one type of drop (ex: monster loot vs tree and rock destruction drops vs pvp loot) or add different configurations for each "type".

**Review by radmint — CHANGES_REQUESTED** [review]

File: ValheimPlus/Drops.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+﻿using HarmonyLib;
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using ValheimPlus.Configurations;
+
+namespace ValheimPlus
+{
+    [HarmonyPatch(typeof(DropTable), "GetDropList", new Type[]
```
[the remark this answers is not in the document]
Comment (Mixone-FinallyHere) [reply ValheimPlus/Drops.cs:9]: Not really

File: ValheimPlus/Drops.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    public static class GetDropList
+    {
+        private static bool Prefix(DropTable __instance, ref List<GameObject> __result)
+        {
+            if (Configuration.Current.Drops.IsEnabled)
+            {
+                int num = UnityEngine.Random.Range(__instance.m_dropMin, __instance.m_dropMax + 1) * Configuration.Current.Drops.baseIncreasedDropMultiplier;
+                __result = __instance.GetDropList(num);
+            }
+            return false;
```
[the remark this answers is not in the document]
Comment (Mixone-FinallyHere) [reply ValheimPlus/Drops.cs:22]: it is making a call to GetDropList
Which is Prefixed

potential iinfinte loop?

**Review by Mixone-FinallyHere — CHANGES_REQUESTED** [review]
Agree with radmint, some changes needed

File: ValheimPlus/Configurations/Sections/DropsConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿namespace ValheimPlus.Configurations.Sections
+{
+    public class DropsConfiguration : ServerSyncConfig<DropsConfiguration>
+    {
+        public int baseIncreasedDropMultiplier { get; internal set; } = 1;
```
[the remark this answers is not in the document]
Comment (gpolcode) [reply ValheimPlus/Configurations/Sections/DropsConfiguration.cs]: I added the multipliers to the `ItemsConfiguration`

File: ValheimPlus/Drops.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gpolcode) [reply ValheimPlus/Drops.cs]: I removed the GetDropList prefix

File: ValheimPlus/Drops.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (gpolcode) [reply ValheimPlus/Drops.cs]: I removed the GetDropList prefix

File: ValheimPlus/Drops.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+    [HarmonyPatch(typeof(Ragdoll), "SpawnLoot")]
```
[the remark this answers is not in the document]
Comment (gpolcode) [reply ValheimPlus/Drops.cs]: I moved both Ragdoll prefixes to the "Ragdoll.cs" file. Should I move the Pickable prefix into another file?

File: ValheimPlus/Drops.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                for (int i = 0; i < @int; i++)
+                {
+                    int int2 = zDO.GetInt("drop_hash" + i, 0);
+                    int value = zDO.GetInt("drop_amount" + i, 0) * Configuration.Current.Drops.baseIncreasedDropMultiplier;
+                    GameObject prefab = ZNetScene.instance.GetPrefab(int2);
+                    list.Add(new KeyValuePair<GameObject, int>(prefab, value));
+                }
+                CharacterDrop.DropItems(list, center + Vector3.up * 0.75f, 0.5f);
+            }
+            return false;
```
[the remark this answers is not in the document]
Comment (gpolcode) [reply ValheimPlus/Drops.cs]: I removed the GetDropList prefix

File: ValheimPlus/Drops.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+using System.Collections.Generic;
+using ValheimPlus.Configurations;
+
+namespace ValheimPlus
+{
+    [HarmonyPatch(typeof(DropTable), "GetDropList", new Type[]
+    {
+
+    })]
+    public static class GetDropList
```
[the remark this answers is not in the document]
Comment (gpolcode) [reply ValheimPlus/Drops.cs]: I removed the GetDropList prefix, but also I tested if I would get additional items when destroying a workbench.

File: ValheimPlus/Drops.cs (hunk 1 above)
Comment (gpolcode) [reply ValheimPlus/Drops.cs]: I split the multiplier into two. Is there any existing way to check for different item types?

**Comment by nxPublic** [comment]
This code is directly copied from another Nexus mod.

We do not search or allow for these kind of "contributions" in this project.

Thank you.

**Pull request closed without merging**